src/utils/cache_manager.js

```diff
@@ -68,7 +68,7 @@ export default class SessionCache {
 
     static putMessages (messages, conversation_id) {
         let sessionMessages = SessionCache.getAllMessages();
-        sessionMessages[conversation_id] = messages;
+        sessionMessages[conversation_id] = JSON.parse(JSON.stringify(messages));
```

Comment:
Why are we stringifying it, just to parse it?

Reply:
It acts as a deep copy. Javascript doesn't really have any other method to do this. I fixed an issue regarding this in the conversation cache, but missed the messages cache.

Reply:
Ahh that's right, I remember this